interfaces/AW-App/src/app/validation-components/upload-data/upload-data.component.ts

```diff
@@ -45,7 +45,22 @@ export class UploadDataComponent implements ValidationComponent {
         if (this.uploadAborted) {
           break;
         }
-        await this.removeCredentialByDid(credential.did);
+        await this.removeLocalStorageData(
+          credential.did,
+          this.ionicStorageTypes.credentials,
+        );
+        await this.removeLocalStorageData(
+          credential.did,
+          this.ionicStorageTypes.validationProgramData,
+        );
+        await this.removeLocalStorageData(
+          credential.did,
+          this.ionicStorageTypes.validationFspData,
+        );
+        await this.removeLocalStorageData(
+          credential.did,
+          this.ionicStorageTypes.qrDidMapping,
+        );
```

Comment:
You don't have to use the local instance of these variables, as they are just string-constants,

```suggestion
        await this.removeLocalStorageData(
          credential.did,
          IonicStorageTypes.credentials,
        );
        await this.removeLocalStorageData(
          credential.did,
          IonicStorageTypes.validationProgramData,
        );
        await this.removeLocalStorageData(
          credential.did,
          IonicStorageTypes.validationFspData,
        );
        await this.removeLocalStorageData(
          credential.did,
          IonicStorageTypes.qrDidMapping,
        );
```